src/test/app/PermissionedDomains_test.cpp

```diff
@@ -0,0 +1,504 @@
+        env(pdomain::setTx(account, pdomain::Credentials(), domain),
+            ter(temMALFORMED));
+
+        // Test 11 credentials.
+        pdomain::Credentials const credentials11{
+            {alice2, "credential1"},
+            {alice3, "credential2"},
+            {alice4, "credential3"},
+            {alice5, "credential4"},
+            {alice6, "credential5"},
+            {alice7, "credential6"},
+            {alice8, "credential7"},
+            {alice9, "credential8"},
+            {alice10, "credential9"},
+            {alice11, "credential10"},
+            {alice12, "credential11"}};
+        BEAST_EXPECT(
+            credentials11.size() ==
+            maxPermissionedDomainCredentialsArraySize + 1);
+        env(pdomain::setTx(account, credentials11, domain), ter(temMALFORMED));
+
+        // Test credentials including non-existent issuer.
+        Account const nobody("nobody");
+        pdomain::Credentials const credentialsNon{
+            {alice2, "credential1"},
+            {alice3, "credential2"},
+            {alice4, "credential3"},
+            {nobody, "credential4"},
+            {alice5, "credential5"},
+            {alice6, "credential6"},
+            {alice7, "credential7"}};
+        env(pdomain::setTx(account, credentialsNon, domain), ter(tecNO_ISSUER));
+
+        // Test bad fee
+        env(pdomain::setTx(account, credentials11, domain),
+            fee(1, true),
+            ter(temBAD_FEE));
+
+        pdomain::Credentials const credentials4{
+            {alice2, "credential1"},
+            {alice3, "credential2"},
+            {alice4, "credential3"},
+            {alice5, "credential4"},
+        };
+        auto txJsonMutable = pdomain::setTx(account, credentials4, domain);
+        auto const credentialOrig = txJsonMutable["AcceptedCredentials"][2u];
+
+        // Remove Issuer from a credential and apply.
+        txJsonMutable["AcceptedCredentials"][2u][jss::Credential].removeMember(
+            jss::Issuer);
+        BEAST_EXPECT(
+            exceptionExpected(env, txJsonMutable).starts_with("invalidParams"));
+
+        txJsonMutable["AcceptedCredentials"][2u] = credentialOrig;
+        // Make an empty CredentialType.
+        txJsonMutable["AcceptedCredentials"][2u][jss::Credential]
+                     ["CredentialType"] = "";
+        env(txJsonMutable, ter(temMALFORMED));
+
+        // Remove Credentialtype from a credential and apply.
+        txJsonMutable["AcceptedCredentials"][2u][jss::Credential].removeMember(
+            "CredentialType");
+        BEAST_EXPECT(
+            exceptionExpected(env, txJsonMutable).starts_with("invalidParams"));
+
+        // Remove both
+        txJsonMutable["AcceptedCredentials"][2u][jss::Credential].removeMember(
+            jss::Issuer);
+        BEAST_EXPECT(
+            exceptionExpected(env, txJsonMutable).starts_with("invalidParams"));
+
+        // Make 2 identical credentials. The duplicate should be silently
+        // removed.
```

Comment:
This comment seems outdated, since it errors.

Reply:
Fixed